tests/liteclient_test.cc

```diff
@@ -210,144 +215,235 @@ class DeviceGatewayMock {
 
 std::string DeviceGatewayMock::RunCmd;
 
+/**
+ * Class LiteClientTest
+ *
+ */
 class LiteClientTest : public ::testing::Test {
  public:
   static std::string SysRootSrc;
 
  protected:
-  LiteClientTest(): sysrootfs_{(test_dir_.Path() / "sysroot-fs").string(), branch, hw_id, os},
-                    sysrepo_{(test_dir_.Path() / "sysrepo").string(), os},
-                    tuf_repo_{test_dir_.Path() / "repo"},
-                    ostree_{(test_dir_.Path() / "treehub").string(), true},
-                    device_gateway_{ostree_, tuf_repo_},
-                    initial_target_{Uptane::Target::Unknown()} {
-
-    auto initial_sysroot_commit_hash = sysrepo_.repo().commit(sysrootfs_.path, sysrootfs_.branch);
-    sysrepo_.deploy(initial_sysroot_commit_hash);
-
-    Json::Value target_json;
-    target_json["hashes"]["sha256"] = initial_sysroot_commit_hash;
-    target_json["custom"]["targetFormat"] = "OSTREE";
-    target_json["length"] = 0;
-    initial_target_ = Uptane::Target{hw_id + "-" + os + "-1", target_json};
+  LiteClientTest()
+      : sys_rootfs_{(test_dir_.Path() / "sysroot-fs").string(), branch, hw_id, os},
+        sys_repo_{(test_dir_.Path() / "sysrepo").string(), os},
+        tuf_repo_{test_dir_.Path() / "repo"},
+        ostree_repo_{(test_dir_.Path() / "treehub").string(), true},
+        device_gateway_{ostree_repo_, tuf_repo_},
+        initial_target_{Uptane::Target::Unknown()},
+        sysroot_hash_{sys_repo_.getRepo().commit(sys_rootfs_.path, sys_rootfs_.branch)} {
+    sys_repo_.deploy(sysroot_hash_);
   }
 
-  std::shared_ptr<LiteClient> create_liteclient(bool initial_version = true) {
+  enum class InitialVersion { kOff, kOn, kCorrupted1, kCorrupted2 };
+
+  /**
+   * method createLiteClient
+   */
+  std::shared_ptr<LiteClient> createLiteClient(InitialVersion initial_version = InitialVersion::kOn) {
     Config conf;
-    conf.uptane.repo_server = deviceGateway().tufRepoUri();
+    conf.uptane.repo_server = getDeviceGateway().getTufRepoUri();
     conf.provision.primary_ecu_hardware_id = hw_id;
     conf.storage.path = test_dir_.Path();
 
     conf.pacman.type = ComposeAppManager::Name;
-    conf.pacman.sysroot = sysrepo_.path();
+    conf.pacman.sysroot = sys_repo_.getPath();
```

Comment:
IMO, `sys_repo_.path()` looks better in terms of readability, and getPath() actually doesn't do any getting it's just an accessor to the object field. Anyway, it's not really important.

Reply:
I understand that but then we should just make the member public...it we are hacking we should at least make it explicit IMO